lib/chef/platform/query_helpers.rb

```diff
@@ -97,6 +99,13 @@ def dsc_refresh_mode_disabled?(node)
         metadata = cmdlet.run!.return_value
         metadata['RefreshMode'] == 'Disabled'
       end
+      
+      def supported_powershell_version?(node, version_string)
+        require 'rubygems'
+        Gem::Version.new(node[:languages][:powershell][:version]) >=
```

Comment:
This will raise `NoMethodError: undefined method `[]' for nil:NilClass` if the powershell ohai plugin didn't find powershell.exe. I see a couple options here. 

* You could `return false unless node[:languages][:powershell]` because no, we don't support that powershell version, because we have no powershell
* Or you could `raise unless node[:languages][:powershell]` because we're considering powershell things and we don't have powershell.

I think the current code paths wouldn't get here, because you have assertions that lead to supports_dsc?, but it seems easy to lose those or other people might call #supported_powershell_version? without ensuring first that powershell existed.